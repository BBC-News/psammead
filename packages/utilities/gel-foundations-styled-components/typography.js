module.exports = require('./dist/typography'); // eslint-disable-line import/no-unresolved
