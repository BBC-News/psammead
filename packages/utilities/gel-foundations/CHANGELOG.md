# Gel Foundations Changelog

| Version | Description |
|---------|-------------|
| 0.1.4   | [PR#224](https://github.com/BBC-News/psammead/pull/224) Add tests for the exported values, coverage 100% :tada: |
| 0.1.1   | [PR#212](https://github.com/BBC-News/psammead/pull/212) Update package description and README. |
| 0.1.0   | [PR#221](https://github.com/BBC-News/psammead/pull/221) Create initial package, pulled in from gel-constants and gel-foundations-styled-components packages. |

