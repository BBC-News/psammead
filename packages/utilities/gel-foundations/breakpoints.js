module.exports = require('./dist/breakpoints'); // eslint-disable-line import/no-unresolved
