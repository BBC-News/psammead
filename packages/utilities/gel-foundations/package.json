{
  "name": "drew-testing-123",
  "version": "1.0.6",
  "main": "dist/index.cjs.js",
  "module": "dist/index.es.js",
  "description": "A range of string constants for use in CSS, intended to help implement BBC GEL-compliant webpages and components.",
  "repository": {
    "type": "git",
    "url": "https://github.com/bbc/psammead/tree/latest/packages/utilities/gel-foundations"
  },
  "author": {
    "name": "Psammead Maintainers",
    "email": "PsammeadMaintainers@bbc.co.uk"
  },
  "files": [
    "dist/"
  ],
  "license": "Apache-2.0",
  "bugs": {
    "url": "https://github.com/bbc/psammead/issues"
  },
  "homepage": "https://github.com/bbc/psammead/blob/latest/packages/utilities/gel-foundations/README.md",
  "keywords": [
    "bbc",
    "utilities",
    "gel",
    "breakpoints",
    "grid",
    "spacing",
    "typography"
  ],
  "devDependencies": {
    "@bbc/psammead-test-helpers": "^1.0.1"
  }
}
