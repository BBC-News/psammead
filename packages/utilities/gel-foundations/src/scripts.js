export latin from './scripts/latin-and-cyrillic';
export cyrillic from './scripts/latin-and-cyrillic';
export latinDiacritics from './scripts/latin-with-diacritics';
export arabic from './scripts/arabic';
export arabicPashto from './scripts/arabic-pashto';
export bengali from './scripts/bengali';
export burmese from './scripts/burmese';
export hindi from './scripts/devanagari-and-gurmukhi';
export nepali from './scripts/devanagari-and-gurmukhi';
export sinhalese from './scripts/sinhalese';
export tamil from './scripts/tamil';
export thai from './scripts/thai';
export chinese from './scripts/no-ascenders-or-descenders';
export korean from './scripts/no-ascenders-or-descenders';
export ethiopic from './scripts/no-ascenders-or-descenders';
