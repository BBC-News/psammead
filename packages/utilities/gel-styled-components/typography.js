import { css } from 'styled-components';
import MEDIA_QUERY_TYPOGRAPHY from './breakpoints';

export const T_LONG_PRIMER = css`
  font-size: 0.9375em;
  line-height: 1.125rem;
  ${MEDIA_QUERY_TYPOGRAPHY.laptopAndLarger} {
    font-size: 0.875em;
  }
`;

/* Font styling below is a subset of BBC GEL Typography "Body Copy" */
export const T_BODY_COPY = css`
  font-size: 0.9375em;
  line-height: 1.25rem;
  ${MEDIA_QUERY_TYPOGRAPHY.smartPhoneAndLarger} {
    font-size: 1em;
    line-height: 1.375rem;
  }
`;

export const T_BREVIER = css`
  font-size: 0.875em;
  line-height: 1rem;
  ${MEDIA_QUERY_TYPOGRAPHY.smartPhoneOnly} {
    line-height: 1.125rem;
  }
  ${MEDIA_QUERY_TYPOGRAPHY.laptopAndLarger} {
    font-size: 0.8125em;
  }
`;

/* Font styling below is a subset of BBC GEL Typography "Canon" */
export const T_CANON = css`
  font-size: 1.75em;
  line-height: 2rem;
  ${MEDIA_QUERY_TYPOGRAPHY.smartPhoneOnly} {
    font-size: 2em;
    line-height: 2.25rem;
  }
  ${MEDIA_QUERY_TYPOGRAPHY.laptopAndLarger} {
    font-size: 2.75em;
    line-height: 3rem;
  }
`;

/* Font styling below is a subset of BBC GEL Typography "Trafalgar" */
export const T_TRAFALGAR = css`
  font-size: 1.25em;
  line-height: 1.5rem;
  ${MEDIA_QUERY_TYPOGRAPHY.smartPhoneOnly} {
    font-size: 1.5em;
    line-height: 1.75rem;
  }
  ${MEDIA_QUERY_TYPOGRAPHY.laptopAndLarger} {
    font-size: 2em;
    line-height: 2.25rem;
  }
`;

export const T_MINION = css`
  font-size: 0.75em;
  line-height: 1em;
`;
