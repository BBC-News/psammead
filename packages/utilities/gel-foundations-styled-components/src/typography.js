import { css } from 'styled-components';
import { MEDIA_QUERY_TYPOGRAPHY } from '@bbc/gel-constants/breakpoints';

export const GEL_LONG_PRIMER = css`
  font-size: 0.9375em;
  line-height: 1.125rem;
  ${MEDIA_QUERY_TYPOGRAPHY.LAPTOP_AND_LARGER} {
    font-size: 0.875em;
  }
`;

/* Font styling below is a subset of BBC GEL Typography "Body Copy" */
export const GEL_BODY_COPY = css`
  font-size: 0.9375em;
  line-height: 1.25rem;
  ${MEDIA_QUERY_TYPOGRAPHY.SMART_PHONE_AND_LARGER} {
    font-size: 1em;
    line-height: 1.375rem;
  }
`;

export const GEL_BREVIER = css`
  font-size: 0.875em;
  line-height: 1rem;
  ${MEDIA_QUERY_TYPOGRAPHY.SMART_PHONE_ONLY} {
    line-height: 1.125rem;
  }
  ${MEDIA_QUERY_TYPOGRAPHY.LAPTOP_AND_LARGER} {
    font-size: 0.8125em;
  }
`;

/* Font styling below is a subset of BBC GEL Typography "Canon" */
export const GEL_CANON = css`
  font-size: 1.75em;
  line-height: 2rem;
  ${MEDIA_QUERY_TYPOGRAPHY.SMART_PHONE_ONLY} {
    font-size: 2em;
    line-height: 2.25rem;
  }
  ${MEDIA_QUERY_TYPOGRAPHY.LAPTOP_AND_LARGER} {
    font-size: 2.75em;
    line-height: 3rem;
  }
`;

/* Font styling below is a subset of BBC GEL Typography "Trafalgar" */
export const GEL_TRAFALGAR = css`
  font-size: 1.25em;
  line-height: 1.5rem;
  ${MEDIA_QUERY_TYPOGRAPHY.SMART_PHONE_ONLY} {
    font-size: 1.5em;
    line-height: 1.75rem;
  }
  ${MEDIA_QUERY_TYPOGRAPHY.LAPTOP_AND_LARGER} {
    font-size: 2em;
    line-height: 2.25rem;
  }
`;

export const GEL_MINION = css`
  font-size: 0.75em;
  line-height: 1em;
`;
