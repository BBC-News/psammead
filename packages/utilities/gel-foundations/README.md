# gel-foundations &middot; [![GitHub license](https://img.shields.io/badge/license-Apache%202.0-blue.svg)](https://github.com/bbc/psammead/blob/latest/LICENSE) [![npm version](https://img.shields.io/npm/v/@bbc/gel-foundations.svg)](https://www.npmjs.com/package/@bbc/gel-foundations) [![PRs Welcome](https://img.shields.io/badge/PRs-welcome-brightgreen.svg)](https://github.com/bbc/psammead/blob/latest/CONTRIBUTING.md)

This package provides a range of string constants for use in CSS, intended to help implement [BBC GEL-compliant](https://www.bbc.co.uk/gel/articles/what-is-gel) webpages and components.

## Exports

`/breakpoints` - GEL breakpoints, as well as typography breakpoints. These use the GEL grid sizes which can be found in the [GEL Grid guidelines](https://www.bbc.co.uk/gel/guidelines/grid#grid-sizes).  
`/spacings` - GEL spacings and GEL Grid margins and gutters. These use the GEL grid spacing sizes which can be found in the [GEL Grid guidelines](https://www.bbc.co.uk/gel/guidelines/grid#spacing-layout).  
`/typography` - GEL typography. These are based on the gel typography standard which can be found in the [GEL guidelines](https://www.bbc.co.uk/gel/guidelines/typography). [More details on how these sizes were implemented for this package are available here.](./typography_sizes_web.md)

## Installation

```jsx
npm install @bbc/gel-foundations --save
```

## Usage

```jsx
import { GEL_GROUP_3_SCREEN_WIDTH_MIN } from '@bbc/gel-foundations/breakpoints';

import { GEL_GUTTER_ABOVE_600PX } from '@bbc/gel-foundations/spacings';

import { GEL_BREVIER } from '@bbc/gel-foundations/typography';
```

These values can then be used directly within CSS declarations in code:

```jsx
import { css } from 'styled-components';

const SomeStyledComponent = css`
  @media (min-width: ${GEL_GROUP_3_SCREEN_WIDTH_MIN}) {
    grid-gap: ${GEL_GUTTER_ABOVE_600PX};
    ${GEL_BREVIER};
  }
`;
```

To allow the typography to be fully accessible and responsive, please note that you should apply a default font-size to the document root (e.g. `html { font-size: 100% }`).

Our typography uses `em` for font-size and `rem` for line-height. `em` allows modularity of components: you can change the component font-size by changing the font-size of its container. `rem` is relative to the document root, so we use that for line-height and spacing for a consistent look-and-feel across the document. You can read our [detailed analysis of "REMs vs EMs for spacing"](https://github.com/bbc/simorgh/blob/latest/docs/Spacing-Units.md) for more information.

## Contributing

When **adding** a new export to this utility package the [export tests](https://github.com/bbc/psammead/blob/5d7395fd60bd8d73796d5a23775b4b5b36db1445/packages/utilities/gel-foundations/index.test.jsx#L13-L59) also need to be updated. When **removing** an exisiting export from this utility package the [export tests](https://github.com/bbc/psammead/blob/5d7395fd60bd8d73796d5a23775b4b5b36db1445/packages/utilities/gel-foundations/index.test.jsx#L13-L59) need to be updated and the package version requires a major change (EG: 1.2.1 -> 2.0.0) as this would be considered a breaking change as functionality is being removed.

Psammead is completely open source. We are grateful for any contributions, whether they be new components, bug fixes or general improvements. Please see our primary contributing guide which can be found at [the root of the Psammead respository](https://github.com/bbc/psammead/blob/latest/CONTRIBUTING.md).

### [Code of Conduct](https://github.com/bbc/psammead/blob/latest/CODE_OF_CONDUCT.md)

We welcome feedback and help on this work. By participating in this project, you agree to abide by the [code of conduct](https://github.com/bbc/psammead/blob/latest/CODE_OF_CONDUCT.md). Please take a moment to read it.

### License

Psammead is [Apache 2.0 licensed](https://github.com/bbc/psammead/blob/latest/LICENSE).
