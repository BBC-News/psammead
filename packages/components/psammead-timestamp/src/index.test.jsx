import React from 'react';
import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';
import { latin } from '@bbc/gel-foundations/scripts';
import { getPica } from '@bbc/gel-foundations/typography';
import Timestamp from '.';

describe('Timestamp', () => {
  shouldMatchSnapshot(
    'should render Timestamp correctly',
    <Timestamp datetime="1530947227000">7 July 2018</Timestamp>,
  );

  shouldMatchSnapshot(
    'should render with the correct typography style applied',
    <Timestamp datetime="1530947227000" typographyFunc={getPica} script={latin}>
      7 July 2018
    </Timestamp>,
  );

  shouldMatchSnapshot(
    'should render Timestamp with a prefix',
    <Timestamp datetime="1530947227000" typographyFunc={getPica} script={latin}>
      Updated 7 July 2018
    </Timestamp>,
  );
});
