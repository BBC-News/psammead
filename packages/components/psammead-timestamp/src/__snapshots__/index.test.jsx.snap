// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timestamp should render Timestamp correctly 1`] = `
.c0 {
  font-size: 0.875rem;
  line-height: 1.125rem;
  color: #6E6E73;
  display: block;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  padding-bottom: 0.25rem;
}

.c0:last-child {
  padding-bottom: 1rem;
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c0 {
    font-size: 0.875rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .c0 {
    font-size: 0.8125rem;
    line-height: 1rem;
  }
}

<time
  className="c0"
  dateTime="1530947227000"
>
  7 July 2018
</time>
`;

exports[`Timestamp should render Timestamp with a prefix 1`] = `
.c0 {
  font-size: 0.9375rem;
  line-height: 1.25rem;
  color: #6E6E73;
  display: block;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  padding-bottom: 0.25rem;
}

.c0:last-child {
  padding-bottom: 1rem;
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c0 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c0 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

<time
  className="c0"
  dateTime="1530947227000"
>
  Updated 7 July 2018
</time>
`;

exports[`Timestamp should render Timestamp without padding 1`] = `
.c0 {
  font-size: 0.9375rem;
  line-height: 1.25rem;
  color: #6E6E73;
  display: block;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c0 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c0 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

<time
  className="c0"
  dateTime="1530947227000"
>
  Updated 7 July 2018
</time>
`;

exports[`Timestamp should render with the correct typography style applied 1`] = `
.c0 {
  font-size: 0.9375rem;
  line-height: 1.25rem;
  color: #6E6E73;
  display: block;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  padding-bottom: 0.25rem;
}

.c0:last-child {
  padding-bottom: 1rem;
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c0 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width:37.5rem) {
  .c0 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

<time
  className="c0"
  dateTime="1530947227000"
>
  7 July 2018
</time>
`;
